### DISCUSSION TITLE
Other options for Filer

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#402] [Ideas]
I was looking for another option for pcmanFM and I found QTFM, but its developer is looking for someone who can accept this project, what do you think?[qtfm github ](https://github.com/rodlie/qtfm)

**Comment by probonopd** [comment]
Do you mean as a file manager instead of Filer? What would be the advantages?

**Comment by grahamperrin** [comment]
I tried the application on FreeBSD 14.0-CURRENT: 

![image](https://user-images.githubusercontent.com/192271/204113641-2dc2fee9-0c66-4e29-8da3-f0398f269a72.png)

First impressions: 

* of the seven devices in the sidebar, six should not be listed (they are not mountable)
* for the one device that could be mounted, it became impossible to unmount the device whilst the application ran.

**Reply by louies0623** [reply]
Does it display all the device sockets of the computer, including the sockets that are not plugged in?

**Comment by louies0623** [comment]
If it's Caja, what are your thoughts?

**Reply by probonopd** [reply]
Gtk is a no-go.

I am thinking about writing Filer 2.0 from scratch in Qt.

**Reply by louies0623** [reply]
You don't have to rewrite, you can modify from QTFM project.

**Reply by probonopd** [reply]
I think it depends on libfm, which is "Gnome in disguise"... we want pure Qt. No Glib. Only C++, no C.

**Comment by louies0623** [comment]
Instead of rewriting a new QT Filer, it is better to use the modify from QTFM project

**Reply by probonopd** [reply]
The point of a clean rewrite would be to not rely on inherited code, be free in the choice of license, etc.

**Reply by louies0623** [reply]
I hope it is as simple and easy to use as the filer of this picture, without the problem of being very confusing because of too many windows.
![Jaguar Server (1)](https://user-images.githubusercontent.com/44593430/206626807-5aadf021-7388-4582-9d13-67d8a66535cd.png)

**Reply by probonopd** [reply]
Please don't paste screenshots of proprietary OSes, thanks. It is ok to say which aspects of which versions of which OSes you like, though.

**Reply by louies0623** [reply]
The shortcut bar of mac os Jaguar filer browses files like a web page can go back or back.

**Reply by louies0623** [reply]
And the downloaded software only needs to be moved to the application icon above, which is very convenient

**Reply by probonopd** [reply]
> The shortcut bar of mac os Jaguar filer browses files like a web page can go back or back.

This is not what we want. We want a spatial Filer not a file "browser".
There is no "back" and "forward" in a spatial file manager. No window is "before" or "after" another window.

https://arstechnica.com/gadgets/2003/04/finder/

https://apple.stackexchange.com/questions/106681/is-there-software-that-implements-john-siracusas-spatial-finder-concept

https://dfeldman.medium.com/about-the-spatial-debate-4ccb8064f1df

**Reply by louies0623** [reply]
When you rewrite it, if possible, I hope it will be like this list.
![image](https://user-images.githubusercontent.com/44593430/208814370-ea70d319-a548-454a-820f-54e179630d70.png)

**Reply by probonopd** [reply]
That seems to be Dino File Manager, a lightweight file manager written in Qt. Maybe we can do some aspects in similar ways. What specifically do you like about it?

https://sourceforge.net/projects/dfm/

**Reply by louies0623** [reply]
List icon mode and function bar with icons, like the picture above

**Comment by louies0623** [comment]
I'm trying to remove the default FM and install xfe to try the desktop icon, but I don't know how to use the command to set xfe as the default desktop icon.
I need help.

**Reply by probonopd** [reply]
I don't know either. helloSystem is deeply integrated with Filer. Filer is not optional and is not intended to be replaced in helloSystem. If you would like another file manager, then helloSystem is probably not a good choice for you. The same goes for Menu and launch. These are the core essentials in helloSystem.
